spmd/tensor/device_mesh.py

```diff
@@ -396,14 +396,35 @@ def all_gather(
         Returns:
             A :class:`torch.Tensor` object
         """
+        my_coordinate = self.get_coordinate_on_dim(mesh_dim)
+        # TODO: what should happen if rank is not in the mesh?
```

Comment:
Let's open an issue / task to consolidate the behavior here?

Reply:
yeah created a issue https://github.com/pytorch/tau/issues/511, will update the comment to link this issue as well